site/snippets/react/useSignTypedData.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+  useSignTypedData,
+  useSmartAccountClient,
+} from "@alchemy/aa-alchemy/react";
+import { useState } from "react";
+
+export function ComponentWithSignTypedData() {
+  // Assumes the app has context of a signer with an authenticated user
+  const [typedDataMessage, setTypedDataMessage] = useState("");
+  const client = useSmartAccountClient({
```

Comment:
this is an example where it would be nice to show the onsuccess or onerror. People would need to use that hook pretty frequently here.

Reply:
could you clarify? you mean the method on this hook to add an onSuccess / onError method?